pkg/smi/client.go

```diff
@@ -188,6 +188,17 @@ func (c *client) ListHTTPTrafficSpecs() []*smiSpecs.HTTPRouteGroup {
 	return httpTrafficSpec
 }
 
+// GetHTTPRouteGroup returns an SMI HTTPRouteGroup resource given its name of the form <namespace>/<name>
+func (c *client) GetHTTPRouteGroup(namespacedName string) *smiSpecs.HTTPRouteGroup {
+	// client-go cache uses <namespace>/<name> as key
+	routeIf, exists, err := c.caches.HTTPRouteGroup.GetByKey(namespacedName)
+	if exists && err == nil {
```

Comment:
not sure, no `IsMonitoredNs`?

Reply:
We don't invoke `Get()` (ex. `GetTCPRoute()`) for objects in un-monitored namespaces, so left it out. Existing use cases always invoke a GET on only monitored objects. Will be dead code if we add that check.

Reply:
Nevetheless, I created a PR to add this check: https://github.com/openservicemesh/osm/pull/3237. If it is a good to have, we can merge that one.